app/models/spree_stripe/address_decorator.rb

```diff
@@ -0,0 +1,17 @@
+module SpreeStripe
+  module AddressDecorator
+    def self.prepended(base)
+      base.after_update :update_stripe_customer, if: :should_update_stripe_customer?
+    end
+
+    delegate :update_stripe_customer, to: :user, allow_nil: true
+
+    private
+
+    def should_update_stripe_customer?
+      user_default_billing? && previous_changes.any?
```

Comment:
We're using the bill address for the Stripe customer so we only need to check that.